extensions/wikia/GlobalNavigation/scripts/GlobalNavigationTracking.js

```diff
@@ -6,39 +6,76 @@ $(function($) {
 	$globalNavigationSearch = $globalNavigation.find('#searchForm');
 	searchSuggestionsShowed = false;
 
+	/**
+	 * Tracking helper function with most commonly used options. Overrides are passed in by callers.
+	 */
+	track = Wikia.Tracker.buildTrackingFunction({
+		category: 'search',
+		trackingMethod: 'both',
+		action: Wikia.Tracker.ACTIONS.CLICK
+	});
+
+	/**
+	 * Parent click handler for events in the global nav
+	 * @param {object} event
+	 */
 	function globalNavigationClickTrackingHandler(event) {
-		var $element, label, $parent;
+		var $element, label;
 
 		//Track only primary mouse button click
 		if (event.type === 'mousedown' && event.which !== 1) {
 			return;
 		}
 
-		track = Wikia.Tracker.buildTrackingFunction({
-			action: Wikia.Tracker.ACTIONS.CLICK,
-			category: 'top-nav',
-			trackingMethod: 'ga'
-		});
-
 		$element = $(event.target);
-		if ($element.closest('.wikia-logo').length > 0) {
-			label = 'wikia-logo';
-		} else if ($element.closest('.hub-links').length > 0 || $element.closest('.hub-list').length > 0) {
+
+		label = getLabelByIdentifier($element) ||
+			getLabelByDOM($element);
+
+		if (label !== false) {
+			track({
+				category: 'top-nav',
+				label: label
+			});
+		}
+	}
+
+	/**
+	 * Get the label for tracking based on data-id attribute
+	 * @param {jQuery} $element
+	 * @returns {string|boolean}
+	 */
+	function getLabelByIdentifier($element) {
+		var identifier = $element.data('id'),
+			map = {
+				'logout': 'user-menu-logout',
+				'help': 'user-menu-help',
+				'preferences': 'user-menu-preferences',
+				'facebook': 'facebook',
+				'register': 'register',
+				'start-wikia': 'start-a-wiki',
+				'wikia-logo': 'wikia-logo',
+				'mytalk': (function () {
+					return $element.hasClass('message-wall') ?
+						'user-menu-message-wall' :
+						'user-menu-talk';
+				})()
+			};
+
+		return map[identifier] || false;
+	}
+
+	/**
+	 * Get the label for tracking based on surrounding DOM of clicked element
+	 * @param {jQuery} $element
+	 * @returns {string|boolean}
+	 */
+	function getLabelByDOM($element) {
+		var label = false,
+			$parent;
+
+		if ($element.closest('.hub-links').length > 0 || $element.closest('.hub-list').length > 0) {
 			label = 'hub-item';
-		} else if ($element.data('id') === 'logout') {
-			label = 'user-menu-logout';
-		} else if ($element.data('id') === 'help') {
-			label = 'user-menu-help';
-		} else if ($element.data('id') === 'preferences') {
-			label = 'user-menu-preferences';
-		} else if ($element.data('id') === 'mytalk') {
-			if ($element.hasClass('message-wall')) {
-				label = 'user-menu-message-wall';
-			} else {
-				label = 'user-menu-talk';
-			}
-		} else if ($element.closest('.ajaxRegisterContainer').length > 0) {
```

Comment:
Good eye, @armonr. This has been replace by this line https://github.com/Wikia/app/pull/5874/files#diff-7fc055b85da28e4e1fe4168f90bb44b1R55. The element inside `.ajaxRegistrationContainer` actually has `data-id=register`, and using the `data-id` value is more performant than calling `.closest()`. Moreover, we don't really want to track clicks on the parent container, which includes padding, we only want to track them on the button itself.